### PR TITLE
Release/1.2.0

### PR DESCRIPTION
**Pull request by JuMastro**
# release:1.2.0

### Breaking changes

- Remove `username` from `TwitchEvents constructor()`, now send it from API functions as argument. 

### Depreciations

- Now set `getGlobalEvents` as deprecated function, recommended to use `getUpcomingEvents` instead.

### Fixes

- Global process rewriting and updates.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,66 +1,47 @@
-# TwitchGetEvents
-[![Build Status](https://travis-ci.org/Lund-Org/twitch-get-events.svg?branch=master)](https://travis-ci.org/Lund-Org/twitch-get-events)
+<h1 align="center"><p>twitch-events</p></h1>
```
[the remark this answers is not in the document]
Comment (JuMastro) [reply README.md]: I prefer to use the `HTML` syntax to centered content.

File: README.md [hunk 2]
```diff
@@ -1,66 +1,47 @@
...
-
-      # If you're an Unix user and you don't have npm
-      docker-compose -f etc/docker/docker-compose.yml up --build dev
-
-      # If you're a Windows user and you have npm
-      npm run windocker
-
-      # If you're a Windows user and you don't have npm
-      docker-compose.exe -f etc/docker/docker-compose.yml up --build dev
+## :arrows_counterclockwise: Running the tests
```
[the remark this answers is not in the document]
Comment (JuMastro) [reply README.md]: You have confirmed the deletion of ./static part. So, docker is not needed again, I think is a good thing for a npm library package.

File: src/index.js [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+    const events = []
+
+    for (const eventCount of offsets) {
+      const startAt = events.length ? events[events.length - 1].startAt : null
+      const requestData = this._makeRequestEventData(username, type, eventCount, startAt)
+      const requestResponse = await this._post(requestOptions, requestData)
+      const requestParsed = this._handleResponse(requestResponse)
+
+      if (!('events' in requestParsed)) {
+        return requestParsed
```
[the remark this answers is not in the document]
Comment (JuMastro) [reply src/index.js]: If `events` property does not exist from the response object, it mean the needed user is not found.

```javascript
// src/index.js (l.257-262)
if (!user.id) {
  return {
    state: IS_FAIL,
    code: USER_NOT_FOUND
  }
}
```

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (FlorianDewulf) [reply README.md]: When I said ok to remove the /static part, we were speaking about tests, so I thought you talked about the tests of the static part (which doesn't make sense). Having an example of how to use the application is interesting 🤔

**Comment by FlorianDewulf** [comment]
@JuMastro 
A little update to add : 
vulnerability
vulnerability

File: static/app.js [hunk 4]
```diff
@@ -1,65 +0,0 @@
-const Hapi = require('hapi')
```
[the remark this answers is not in the document]
Comment (JuMastro) [reply static/app.js]: It's back

[the remark this answers is not in the document]
Comment (JuMastro) [reply static/app.js]: done

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (JuMastro) [reply README.md]: done

File: static/app.js [hunk 5]
```diff
@@ -13,11 +13,10 @@ server.route({
   path: '/upcoming/{clientId}/{username}',
```
Comment (Mystilund) [inline static/app.js]: clientId remaining in url but not use anymore

File: static/app.js [hunk 6]
```diff
@@ -29,9 +28,8 @@ server.route({
   method: 'GET',
   path: '/past/{clientId}/{username}',
```
Comment (Mystilund) [inline static/app.js]: same as previous review

**Pull request closed without merging**